halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/ha/HaServiceRedirectsProfileFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile.ha;
+
+import com.netflix.spinnaker.halyard.config.model.v1.node.DeploymentConfiguration;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.SpinnakerRuntimeSettings;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile.Profile;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.profile.StringBackedProfileFactory;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.ServiceSettings;
+import com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.SpinnakerService.Type;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Data;
+import org.springframework.stereotype.Component;
+
+@Component
+public abstract class HaServiceRedirectsProfileFactory extends StringBackedProfileFactory {
+  protected static final ServiceSettings CLOUDDRIVER_TO_CLOUDDRIVER_RO_REDIRECT = ServiceSettings.serviceBaseUrlRedirect("${services.clouddriver-ro.baseUrl:http://localhost:7002}");
```

Comment:
I wouldn't encode the actual endpoints here. I would make the desired endpoint an input of this function (as a SpinnakerService object likely) decoupling the endpoint format from this class. For example, this will immediately break if someone deploys to a namespace other than "spinnaker".